src/controllers/scripting/legacy/controllerscriptenginelegacy.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <QFileSystemWatcher>
+#include <QJSEngine>
+#include <QJSValue>
+#include <QMessageBox>
+
+#include "controllers/controllerpreset.h"
+#include "controllers/scripting/controllerscriptenginebase.h"
+#include "util/duration.h"
+
+class Controller;
+class ControllerScriptInterfaceLegacy;
+class EvaluationException;
+class ScriptConnection;
+
+/// ControllerScriptEngineLegacy loads and executes controller scripts for the legacy
+/// JS/XML hybrid controller mapping system.
+class ControllerScriptEngineLegacy : public ControllerScriptEngineBase {
+    Q_OBJECT
+  public:
+    explicit ControllerScriptEngineLegacy(Controller* controller);
+    ~ControllerScriptEngineLegacy() override;
+
+    bool initialize() override;
+
+    bool handleIncomingData(const QByteArray& data);
+
+    /// Wrap a string of JS code in an anonymous function. This allows any JS
+    /// string that evaluates to a function to be used in MIDI mapping XML files
+    /// and ensures the function is executed with the correct 'this' object.
+    QJSValue wrapFunctionCode(const QString& codeSnippet, int numberOfArgs);
+
+    // There is lots of tight coupling between ControllerScriptEngineLegacy
+    // and ControllerScriptInterface. This is probably not worth improving in legacy code.
+    QJSEngine* jsEngine() {
+        return m_pJSEngine;
+    }
+
+  public slots:
+    void setScriptFiles(const QList<ControllerPreset::ScriptFileInfo>& scripts) {
+        m_fileWatcher.removePaths(m_fileWatcher.files());
+        m_scriptFiles = scripts;
+    }
+
+  private:
+    bool evaluateScriptFile(const QFileInfo& scriptFile);
+    void shutdown() override;
+
+    void generateScriptFunctions(const QString& code);
```

Comment:
Functions with only a side-effect should be avoided, even as member functions.

Reply:
Apparently this function isn't even defined anyway, so I just removed it.